tensorflow/compiler/xla/service/buffer_assignment.cc

```diff
@@ -1618,7 +1638,9 @@ StatusOr<std::unique_ptr<BufferAssignment>> BufferAssigner::CreateAssignment(
   // This can only be performed after all buffers have been assigned, and
   // after maybe_live_out is marked, since it is used to determine whether an
   // allocation contains temporary buffers or not.
-  assignment->CombineTempAllocations();
+  if (multiheap_size_constraint_per_heap == -1) {
+    assignment->CombineTempAllocations();
```

Comment:
Why temp allocations are not combined in constrained case? What if their size is below the constraint?

Reply:
Because we want multiple temp allocations. If we combine them, they will become just one large allocation.